config-model/src/main/java/com/yahoo/vespa/configmodel/producers/DocumentManager.java

```diff
@@ -176,4 +196,271 @@ private void buildImportedFieldsConfig(Collection<String> fieldNames, Datatype.D
         }
     }
 
+
+    // Alternate (new) way to build config:
+
+    public DocumentmanagerConfig.Builder produceDocTypes(DocumentModel model, DocumentmanagerConfig.Builder builder) {
+        builder.usev8geopositions(this.useV8GeoPositions);
+        Map<NewDocumentType.Name, NewDocumentType> produced = new HashMap<>();
+        var indexMap = new IdxMap();
+        for (NewDocumentType documentType : model.getDocumentManager().getTypes()) {
+            docTypeInheritOrder(documentType, builder, produced, indexMap);
+        }
+        indexMap.verifyAllDone();
+        return builder;
+    }
+
+    private void docTypeInheritOrder(NewDocumentType documentType,
+                                     DocumentmanagerConfig.Builder builder,
+                                     Map<NewDocumentType.Name, NewDocumentType> produced,
+                                     IdxMap indexMap)
+    {
+        if (! produced.containsKey(documentType.getFullName())) {
+            for (NewDocumentType inherited : documentType.getInherited()) {
+                docTypeInheritOrder(inherited, builder, produced, indexMap);
+            }
+            docTypeBuild(documentType, builder, indexMap);
+            produced.put(documentType.getFullName(), documentType);
+        }
+    }
+
+    static private class IdxMap {
+        private Map<Integer, Boolean> doneMap = new HashMap<>();
+        private Map<Object, Integer> map = new IdentityHashMap<>();
```

Comment:
If you are using IdentityHashMap I assume that it is not used for iteration where order matters.